server/routes/manageConditions/handlers/curfew/curfewRoutes.ts

```diff
@@ -0,0 +1,116 @@
+import { Request, Response } from 'express'
+import LicenceService from '../../../../services/licenceService'
+import ConditionService from '../../../../services/conditionService'
+import { AddAdditionalConditionRequest, Licence } from '../../../../@types/licenceApiClientTypes'
+import { SimpleTime } from '../../types'
+import { User } from '../../../../@types/CvlUserDetails'
+
+export default class CurfewRoutes {
+  constructor(
+    private readonly licenceService: LicenceService,
+    private readonly conditionService: ConditionService
+  ) {}
+
+  GET = async (req: Request, res: Response): Promise<void> => {
+    const { conditionCode } = req.params
+    const { licence } = res.locals
+
+    const additionalCondition = licence.additionalLicenceConditions.find(condition => condition.code === conditionCode)
+    const conditionInstances = licence.additionalLicenceConditions
+      .filter(condition => condition.code === conditionCode)
+      .sort((a, b) => a.id - b.id)
+    const config = await this.conditionService.getAdditionalConditionByCode(conditionCode, licence.version)
+
+    if (!additionalCondition) {
+      return res.redirect(
+        `/licence/create/id/${licence.id}/additional-licence-conditions${
+          req.query?.fromReview ? '?fromReview=true' : ''
+        }`
+      )
+    }
+
+    const formResponses = conditionInstances.reduce((acc, instance, index) => {
+      instance.data.map(conditionData => {
+        const key = index === 0 ? conditionData.field : `${conditionData.field}${index + 1}`
+        acc[key] = conditionData.value
+        return conditionData.value
+      })
+      return acc
+    }, {})
+
+    return res.render('pages/manageConditions/curfew/input', { additionalCondition, config, formResponses })
+  }
+
+  POST = async (req: Request, res: Response): Promise<void> => {
+    const { conditionCode } = req.params
+    const { user, licence } = res.locals
+    const inputs = req.body
+
+    await this.licenceService.deleteAdditionalConditionsByCode(conditionCode, licence, user)
+
+    if (inputs.numberOfCurfews === 'Two curfews') {
+      await this.addCurfewCondition(licence, user, conditionCode, inputs.curfewStart, inputs.curfewEnd, inputs)
+      await this.addCurfewCondition(licence, user, conditionCode, inputs.curfewStart2, inputs.curfewEnd2, inputs)
+    } else if (inputs.numberOfCurfews === 'Three curfews') {
+      await this.addCurfewCondition(licence, user, conditionCode, inputs.curfewStart, inputs.curfewEnd, inputs)
+      await this.addCurfewCondition(licence, user, conditionCode, inputs.curfewStart2, inputs.curfewEnd2, inputs)
+      await this.addCurfewCondition(licence, user, conditionCode, inputs.curfewStart3, inputs.curfewEnd3, inputs)
+    } else {
+      await this.addCurfewCondition(licence, user, conditionCode, inputs.curfewStart, inputs.curfewEnd, inputs)
+    }
```

Comment:
The reverse to the GET method, where a single page of inputs potentially needs to be split into multiple instances of the condition.
I thought about making a method that parsed the selected number of curfews, or relied on null/undefined checks to determine how many instances of the condition to add, but in the end I thought it's a minor efficiency saving at the cost of a significant amount of readability.

That said, I'm happy to refactor into a more succinct method if desired.

Reply:
Ideally we'd just have a `replace` endpoint on the backend..


I guess you could have done:
```ts
await this.addCurfewCondition(licence, user, conditionCode, inputs.curfewStart, inputs.curfewEnd, inputs)

if (two or three) {
   await this.addCurfewCondition(licence, user, conditionCode, inputs.curfewStart2, inputs.curfewEnd2, inputs)
}

if (three) {
     await this.addCurfewCondition(licence, user, conditionCode, inputs.curfewStart3, inputs.curfewEnd3, inputs)
}
```